gix-actor/src/signature/decode.rs

```diff
@@ -68,31 +68,29 @@ pub(crate) mod function {
     /// Parse an identity from the bytes input `i` (like `name <email>`) using `nom`.
     pub fn identity<'a, E: ParserError<&'a [u8]> + AddContext<&'a [u8], StrContext>>(
         i: &mut &'a [u8],
-    ) -> PResult<IdentityRef<'a>, E> {
+    ) -> ModalResult<IdentityRef<'a>, E> {
         let start = i.checkpoint();
         let eol_idx = i.find_byte(b'\n').unwrap_or(i.len());
-        let right_delim_idx =
-            i[..eol_idx]
-                .rfind_byte(b'>')
-                .ok_or(ErrMode::Cut(E::from_error_kind(i, ErrorKind::Eof).add_context(
-                    i,
-                    &start,
-                    StrContext::Label("Closing '>' not found"),
-                )))?;
+        let right_delim_idx = i[..eol_idx]
```

Comment:
This change was introduced in https://github.com/GitoxideLabs/gitoxide/pull/1439 and I wonder if this there isn't an 'packagiomatic' way of writing this. This code implements the parser by hand, but that was done merely due to lack of knowledge on how to use `winnow` primitives.

Is there anything that could be done to make it nicer?

Reply:
The main challenge is the `rfind`, right?  I've not thought of a good way to handle cases like that.

If its parsing within the `\n` delimited range, there is `and_then`, like `take_until(0.., b'\n').and_then(parser)`.

Reply:
Yes, it's the `rfind()` to assure it finds the right-most `>` as delimiter, within the line boundary.